File: nlp_backend/backend/backend.js
const express = require('express');
const cors = require('cors');
const pickle = require('pickle');
const { CountVectorizer } = require('scikit-learn');
const { pythonShell } = require('python-shell');
var bodyParser = require('body-parser')
 
var app = express()
 
// create application/json parser
var jsonParser = bodyParser.json()


// Annoying CORS
app.use(cors());

const labels = {
  0: 'anger',
  1: 'fear',
  2: 'happiness',
  3: 'love',
  4: 'sadness',
};

class DiaryText {
  constructor(text) {
    this.text = text;
  }
}

app.get('/', (req, res) => {
  res.json({ message: 'Please refer to the README for more information.' });
});

app.post('/predict', jsonParser, (req, res) => {
    console.log(req.body);
  const text = new DiaryText(req.body.text);
  const vector = pickle.loads('nlp_backend/vector.pickle');
  const model = pickle.loads('nlp_backend/model.pickle');
  const textVector = vector.transform([text]);
  model.predict(textVector).then((yPred) => {
    res.json({ category: labels[yPred[0]] });
  });
});

app.listen(8000, () => {
  console.log('Server started on http://localhost:8000');
});

File: nlp_backend/backend/backend.py
import pickle
from pydantic import BaseModel
from fastapi import FastAPI
from sklearn.feature_extraction.text import CountVectorizer
from fastapi.middleware.cors import CORSMiddleware


app = FastAPI()

# Annoying CORS
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

labels = {
    0: "anger",
    1: "fear",
    2: "happiness",
    3: "love",
    4: "sadness",
}


class DiaryText(BaseModel):
    text: str


@app.get("/")
def home():
    return {"message": "Please refer to the README for more information."}


@app.post("/predict")
async def predict(text: DiaryText):

    vector = pickle.load(open("nlp_backend/vector.pickle", 'rb'))
    model = pickle.load(open("nlp_backend/model.pickle", 'rb'))
    text = str(text)

    text_vector = vector.transform([text])
    y_pred = model.predict(text_vector)

    return {"category": labels[y_pred[0]]}
